flybywiresim/aircraft
# feat(a380x): us units option

Fixes #9206 

## Summary of Changes


- Added conversion for FUEL&LOAD page entries.
- Displaying values on Fuel/Apu/Engine/Cruise SD pages in the correct unit
- Displaying correct unit on OANS page
- Displaying correct unit for T.O Shift on PERF TO page
- Displaying correct unit on F-Plan for Departure/approach runway
- Fixed some UI components to be more efficient and a lot easier to use.
- Fixed some type issues with NXDataStore.
- Converted CONFIG_USING_METRIC to the new style setting.

## Screenshots (if necessary)



## References


![image](https://github.com/user-attachments/assets/b3416eba-7080-4615-bdbe-43cd89593e52)




## Additional context



Discord username (if different from GitHub): Mikio

## Testing instructions


- Switch weight unit on Flypad Aircraft setting, enter values as LBS on Fuel & Load page. Cycle through SD pages of APU/Engine/Fuel/Cruise to check for correct unit of values.
- Select Departure Runway, Select Arrival Runway on F-PLAN and check it shows correct unit.
- Use OANS and BTV, and select runway and exit and check it shows correct unit.


## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo**, **flybywire-aircraft-a380-842 (4K)** or **flybywire-aircraft-a380-842 (8K)** download link at the bottom of the page
